Vikta-API-testing-sample/src/test/java/com/test/api/user/UserTest.java

```diff
@@ -0,0 +1,195 @@
+        assertThat(users).as("User don't have loginName field").allMatch(user -> !user.getLoginName().isEmpty());
+        assertThat(users).as("User don't have password field").allMatch(user -> !user.getPassword().isEmpty());
+        assertThat(users).as("User don't have email field").allMatch(user -> !user.getEmail().isEmpty());
+        assertThat(users).as("User don't have firstName field").allMatch(user -> !user.getFirstName().isEmpty());
+        assertThat(users).as("User don't have middleName field").allMatch(user -> !user.getMiddleName().isEmpty());
+        assertThat(users).as("User don't have surname field").allMatch(user -> !user.getSurname().isEmpty());
+    }
+
+    @Test
+    public void canGetUserByLoginName(){
+        final String loginPath = getTestEnvironment().getUserPath() + "/login";
+        final String loginQueryParam = "login";
+
+        List<UserDTO> users = getListOfUsers();
+        String userLogin = users.get(new Random().nextInt(users.size())).getLoginName();
+
+        List<UserDTO> user = given().spec(defaultRequestSpec())
+                .queryParam(loginQueryParam, userLogin)
+                .when()
+                .log().all()
+                .get(loginPath)
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract()
+                .body().jsonPath().getList(".", UserDTO.class);
+
+        assertThat(user.size())
+                .as("There should be no users with same login")
+                .isLessThan(2);
+
+        assertThat(user.get(0).getLoginName())
+                .as(String.format("Should be user with login %s, but was with %s", userLogin, user.get(0).getLoginName()))
+                .isEqualTo(userLogin);
+    }
+
+    // when user is not found (does not exist) returns empty body, I think that better will be returned code 404 - not found
+    @Test
+    public void canNotGetUserByWrongLoginName(){
+        final String loginPath = getTestEnvironment().getUserPath() + "/login";
+        final String loginQueryParam = "login";
+        final String userLogin = "ThereIsNotUserWithThatLogin";
+
+        List<UserDTO> user = given().spec(defaultRequestSpec())
+                .queryParam(loginQueryParam, userLogin)
+                .when()
+                .log().all()
+                .get(loginPath)
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract()
+                .body().jsonPath().getList(".", UserDTO.class);
+
+        assertThat(user).isEmpty();
+    }
+
+    @Test
+    public void canSearchUserByTerm(){
+        final String searchPath = getTestEnvironment().getUserPath() + "/search";
+        final String termQueryParam = "term";
+        Random random = new Random();
+        List<UserDTO> users = getListOfUsers();
+
+        assertThat(users.size()).isGreaterThan(0);
+
+        int size = random.nextInt(users.size()-1) + 2;
```

Comment:
you don't have to substract 1. nextInt method picks a value between 0 (inclusive) and the given bound (exclusive)